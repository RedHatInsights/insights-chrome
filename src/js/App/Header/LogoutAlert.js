import React from 'react';
import { Alert, AlertActionCloseButton } from '@patternfly/react-core';
import cookie from 'js-cookie';

class LogoutAlert extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            alertOneVisible: cookie.get('cs_loggedOut') === 'true',
            delay: 8000
        };
    }

    hideAlertOne = () => {
        this.setState({ alertOneVisible: false });
        cookie.set('cs_loggedOut', 'false');
    }
    componentDidMount() {
        this.setTimer();
    }
    componentWillUnmount() {
        clearTimeout(this._timer);
    }

    setTimer = () => {
        if (this._timer !== null) {
            clearTimeout(this._timer);
        }

        // hide after `delay` milliseconds
        this._timer = setTimeout(function() {
            this.setState({ alertOneVisible: false });
            clearTimeout(this._timer);
        }.bind(this), this.state.delay);
    }

    render() {
        const { alertOneVisible } = this.state;
        return (
            <React.Fragment>
                { alertOneVisible && (
                    <Alert
                        variant="success"
                        title="You have successfully logged out."
                        className="ins-c-alert__logout"
                        action={ <AlertActionCloseButton onClose={ this.hideAlertOne } /> }
                    >
                    </Alert>
                ) }
            </React.Fragment>
        );
    }
}

export default LogoutAlert;
