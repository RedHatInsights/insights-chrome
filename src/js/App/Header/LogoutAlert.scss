.ins-c-alert__logout {
    .pf-c-alert__title {
        padding: unset;
    }

    .pf-c-alert__icon {
        padding: unset;
        background-color: unset;
    }
}