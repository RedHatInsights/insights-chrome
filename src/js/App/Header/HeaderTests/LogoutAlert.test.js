import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import LogoutAlert from '../LogoutAlert';

describe('Login', () => {

    it('should render correctly with no content', () =>{
        Object.defineProperty(window.document, 'cookie', {
            writable: true,
            value: `cs_loggedOut=false`
        });
        const wrapper = shallow(<LogoutAlert />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render correctly with content', () =>{
        Object.defineProperty(window.document, 'cookie', {
            writable: true,
            value: `cs_loggedOut=true`
        });
        const mockClose = jest.fn();
        const wrapper = shallow(<LogoutAlert onClose={ mockClose } />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
