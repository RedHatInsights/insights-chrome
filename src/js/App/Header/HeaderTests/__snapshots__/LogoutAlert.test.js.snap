// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should render correctly with content 1`] = `
<Fragment>
  <Component
    action={
      <Unknown
        onClose={[Function]}
      />
    }
    className="ins-c-alert__logout"
    title="You have successfully logged out."
    variant="success"
  />
</Fragment>
`;

exports[`Login should render correctly with no content 1`] = `<Fragment />`;
